### PR TITLE
Use Rasabot token

### PR DESCRIPTION
**Pull request by HotThoughts**
**The problem**

PRs created by `GITHUB_TOKEN` do not trigger any workflow runs. 

Examples: 
- https://github.com/RasaHQ/rasa-x-helm/pull/213
- https://github.com/RasaHQ/rasa-x-helm/pull/214

> When you use the repository's `GITHUB_TOKEN` to perform tasks on behalf of the GitHub Actions app, events triggered by the GITHUB_TOKEN will not create a new workflow run. 
> Ref: https://docs.github.com/en/actions/reference/authentication-in-a-workflow

**Proposed changes**

Replace all `GITHUB_TOKEN` secret to Rasabot.

### CODE REVIEW

File: .github/workflows/changelog.yml [hunk 1]
```diff
@@ -27,15 +27,15 @@ jobs:
       if: startsWith(steps.changelog.outputs.changelog, '####')
       run: |
         # Get release id
-        RELEASE_ID=$(curl -s -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
+        RELEASE_ID=$(curl -s -H "Authorization: token ${{ secrets.RASABOT_GITHUB_TOKEN }}" \
```
Comment (tczekajlo) [inline .github/workflows/changelog.yml:30]: Why did we change it here?

File: .github/workflows/changelog.yml [hunk 2]
```diff
@@ -27,15 +27,15 @@ jobs:
...
           -H "Accept: application/vnd.github.v3+json" \
           "https://api.github.com/repos/${{ github.repository }}/releases/latest" | jq '.id')
 
         RELEASE_BODY="${{ steps.changelog.outputs.changelog }}"
         RELEASE_BODY="${RELEASE_BODY//$'\n'/'\n'}"
 
         # Update release body
-        curl -X PATCH -s -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
+        curl -X PATCH -s -H "Authorization: token ${{ secrets.RASABOT_GITHUB_TOKEN }}" \
```
Comment (tczekajlo) [inline .github/workflows/changelog.yml:38]: And here?

File: .github/workflows/changelog.yml (hunk 1 above)
Comment (HotThoughts) [reply .github/workflows/changelog.yml:30]: Thanks for pointing this out! Right we could use `secrets.GITHUB_TOKEN` here as it has enough permission in this case. And it is also safer to use`secrets.GITHUB_TOKEN`. Will revert it back to `secrets.GITHUB_TOKEN`.

**Review by tczekajlo — APPROVED** [review]
LGTM

**Pull request merged**